projects/helper/chain/radixdlt.js

```diff
@@ -40,45 +40,74 @@ async function queryAddresses({ addresses = [], }) {
   return items
 }
 
-async function queryLsus(addresses = []) {
-  // Receives a list of lsu addresses and returns the redemption value in XRD
-
+async function queryLiquidStakeUnitDetails(addresses = []) {
+  /**
+   * queryLsus takes a list of addresses as argument.
+   * The return value is a dictionary with the lsu address and it's valuation in xrd.
+   * Example response:
+   *
+   * {
+      resource_rdxabc1: {
+        totalSupplyOfStakeUnits: '48921317.41255863564527699',
+        validatorAddress: 'validator_rdx11234',
+        xrdRedemptionValue: 1.0087202808923446
+      },
+      resource_rdx1cde1: {
+        totalSupplyOfStakeUnits: '27022271.985429161792201709',
+        validatorAddress: 'validator_rdx15678',
+        xrdRedemptionValue: 1.0087163411719327
+      }
+    }
+  */
   let lsuRedemptionValues = {}
   const chunks  = sliceIntoChunks(addresses, 20)
   for (const chunk of chunks) {
     let body = {
       "addresses": chunk,
-      "aggregation_level": "Vault"
+      "aggregation_level": "Vault",
+      "explicit_metadata": ["validator"]
     }
-    let data = await post(ENTITY_DETAILS_URL, body)
+    try {
+      let data = await post(ENTITY_DETAILS_URL, body)
 
-    let validators = []
-    for (const lsuResource of data.items) {
-      let validator = lsuResource.metadata.items.filter(metadataItem => metadataItem.key === "validator")[0]
-      lsuRedemptionValues[lsuResource.address] = {
-        "totalSupplyOfStakeUnits": lsuResource.details.total_supply,
-        "validatorAddress": validator.value.typed.value
-      }
+      let validators = []
+      for (const lsuResource of data.items) {
+        let validator = lsuResource.metadata.items.filter(metadataItem => metadataItem.key === "validator")[0]
+        if (validator !== undefined) {
```

Comment:
Based on my admittedly slightly rust JS experience, I don't _think_ this check is quite right - I think you should check for undefined before you index with `[0]` else it would be a `TypeError` on line 75.

Reply:
true.